packages/ui/src/forum/modals/CreateThreadModal/CreateThreadModal.tsx

```diff
@@ -0,0 +1,39 @@
+import { useMachine } from '@xstate/react'
+import React, { useEffect } from 'react'
+
+import { useModal } from '@/common/hooks/useModal'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+
+import { SwitchMemberModalCall } from '../../../memberships/modals/SwitchMemberModal'
+
+import { CreateThreadDetailsModal } from './CreateThreadDetailsModal'
+import { createThreadMachine } from './machine'
+
+export const CreateThreadModal = () => {
+  const { active: member } = useMyMemberships()
+  const { hideModal, showModal } = useModal()
+  const [state, send] = useMachine(createThreadMachine)
+
+  useEffect(() => {
+    if (state.matches('requirementsVerification')) {
+      !member ? showModal<SwitchMemberModalCall>({ modal: 'SwitchMember' }) : send('NEXT')
```

Comment:
Please use if-else if there's an action. That's because, in most cases, ternary operator would be used in assignement.